src/server/plugins/engine/views/common/warn-missing-notification-email.html

```diff
@@ -0,0 +1,11 @@
+{% from "govuk/components/warning-text/macro.njk" import govukWarningText %}
+
+{% macro warnMissingNotificationEmail(notificationEmail, designerUrl, slug) %}
+  {% if not notificationEmail | length %} 
+    {{ govukWarningText({
+        html: 'If you want to receive a test email, <a target="_blank" class="govuk-link" href=' + designerUrl + '/library/' + slug + '/edit/notification-email>enter the email address (opens in new tab)</a> you want form submissions to go to.',
+        iconFallbackText: "Warning"
+    }) }}
+  {% endif %}
+{% endmacro %}
```

Comment:
changing macro to include is also ok but I think I prefer this. Let me know what you think!

Reply:
Nice one @KiranVarma02

We'll need to restrict this to "preview mode" otherwise there's a possibility our link to **forms-designer** will appear to users in production on live-but-misconfigured forms

For your macro versus partial question, shall we keep it as a partial like the others?

We've got `slug` etc available via the default context so they don't need to be macro params:

https://github.com/DEFRA/forms-runner/blob/9ba19cb625ebfc7b56b220c6a722f9667a7b5880/src/server/plugins/views.ts#L66-L72

Reply:
You might like what we do in **forms-designer**

We set up a cut-down [`config` context property](https://github.com/DEFRA/forms-designer/blob/c7dd495cd4f9eb8cefd95d175648527100cb6d97/designer/server/src/common/nunjucks/context/index.js#L35-L38) which gives `{{ config.serviceName }}` in Nunjucks